src/dqrobotics/interfaces/vrep/DQ_VrepInterface.cpp

```diff
@@ -566,3 +590,531 @@ bool DQ_VrepInterface::is_video_recording()
+ *              //
+ *              //  function get_center_of_mass(inInts,inFloats,inStrings,inBuffer)
+ *              //     if #inInts>=1 then
+ *              //         IM,matrix_SF_COM=sim.getShapeInertia(inInts[1])
+ *              //         matrix0_SF=sim.getObjectMatrix(inInts[1],-1)
+ *              //         if inStrings[1] == 'absolute_frame' then
+ *              //             resultMatrix = sim.multiplyMatrices(matrix0_SF,matrix_SF_COM)
+ *              //         else
+ *              //             resultMatrix = matrix_SF_COM
+ *              //         end
+ *              //         return {},{resultMatrix[4], resultMatrix[8],resultMatrix[12]},{},''
+ *              //     end
+ *              // end
+ *              // In addition, it is assumed a FrankaEmikaPanda robot manipulator
+ *              // in the Coppelia scene.
+ *
+ *              DQ_VrepInterface vi;
+ *              MatrixXd inertia_matrix = vi.extract_center_of_mass("DQRoboticsApiCommandServer","get_inertia","Franka_link2_resp");
+ *
+ */
+VectorXd DQ_VrepInterface::extract_center_of_mass(const std::string&  obj_name, const std::string&  function_name, const std::string& link_name)
+{
+    struct call_script_data data;
+    int my_handle = get_object_handle(link_name);
+    data = call_script_function(obj_name, function_name, {my_handle}, {}, {});
+    int size = data.output_floats.size();
+    if (size != 3){
+        throw std::range_error("Error in extract_center_of mass. Incorrect number of returned values from CoppeliaSim. (Expected: 3)");
+    }
+    VectorXd center_of_mass = VectorXd::Zero(3);
+    center_of_mass << data.output_floats[0],data.output_floats[1],data.output_floats[2];
+    return center_of_mass;
+}
+
+
+/**
+ * @brief This method returns the mass of an object on the CoppeliaSim scene.
+ * @param obj_name The name of the object where the script is attached to.
+ * @param function_name The name of the script function to call in the specified script.
+ * @param link name. The name of the object from which we want to extract the mass.
+ * @returns The mass of the object.
+ *
+ *              Example:
+ *              // This example assumes that in your CoppeliaSim there is a child script called
+ *              // "DQRoboticsApiCommandServer", where is defined the following Lua function:
+ *              //
+ *              //  function get_mass(inInts,inFloats,inStrings,inBuffer)
+ *              //     local mass = {}
+ *              //     if #inInts>=1 then
+ *              //         mass =sim.getShapeMass(inInts[1])
+ *              //         return {},{mass},{},''
+ *              //     end
+ *              //  end
+ *              // In addition, it is assumed that there is a FrankaEmikaPanda robot manipulator
+ *              // in the Coppelia scene.
+ *
+ *              DQ_VrepInterface vi;
+ *              double mass = vi.extract_mass("DQRoboticsApiCommandServer","get_mass","Franka_link2_resp");
```

Comment:
`get_mass` and `extract_mass` ?

Reply:
I modified the name of all three methods. Now, I use "get" instead of "extract".